lib/active_merchant/billing/gateways/adyen.rb

```diff
@@ -982,6 +981,17 @@ def unsupported_failure_response(initial_response)
       def card_not_stored?(response)
         response.authorization ? response.authorization.split('#')[2].nil? : true
       end
+
+      def skip_mpi_data?(options = {})
+        # Skips adding the NT mpi data if it is explicitly skipped in options, or if it is MIT and not the initial transaction.
+        options[:skip_mpi_data] == 'Y' || (!options.dig(:stored_credential, :initial_transaction) && options.dig(:stored_credential, :initiator) == 'merchant' && options[:switch_cryptogram_mapping_nt])
+      end
+
+      def ecommerce_shopper_interaction?(payment, options)
+        (options.dig(:stored_credential, :initial_transaction) && options.dig(:stored_credential, :initiator) == 'cardholder') ||
+          (payment.respond_to?(:verification_value) && payment.verification_value && options.dig(:stored_credential, :initial_transaction)) ||
+          (payment.is_a?(NetworkTokenizationCreditCard) && !options[:switch_cryptogram_mapping_nt])
```

Comment:
Does this mean you want to pass ecommerce if this a NetworkTokenizationCreditCard and switch_cryptogram_mapping_nt is nil?

Reply:
Yeah if it is nil or false. 

I did this because when the flag is on I want to remove the payment.is_a?(NetworkTokenizationCreditCard) check that is there currently, and this allows the current functionality to stay the same while the flag is off.